File: csharp/Exceptions/AuthenticationException.cs
using System;

namespace Authdog.Exceptions
{
    /// <summary>
    /// Exception thrown when authentication fails
    /// </summary>
    public class AuthenticationException : AuthdogException
    {
        public AuthenticationException(string message = "Authentication failed") : base(message)
        {
        }

        public AuthenticationException(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}

File: planned/php/src/Exceptions/AuthenticationException.php
<?php

namespace Authdog\Exceptions;

/**
 * Exception thrown when authentication fails
 */
class AuthenticationException extends AuthdogException
{
    public function __construct(string $message = "Authentication failed", int $code = 401, ?\Throwable $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}
